### PR TITLE
Improved theme structure document

### LINKED ISSUE #3164: Feedback on page: /guides/v2.2/frontend-dev-guide/themes/theme-structure.html
# General issue

## Description:

Directory | Required | Description
-- | -- | --
/<Vendor>_<Module>/web/css/source | optional | Module-specific styles (.css and/or .less files). General styles for the module are in the _module.less file, and styles for widgets are in _widgets.less.

it should be <Vendor>_<Module>/web/css/ as .css files are not under /source ex: [Module_Theme](https://github.com/magento/magento2/tree/2.3-develop/app/code/Magento/Theme/view/frontend/web/css)  or description field should be corrected

Directory | Required | Description
-- | -- | --
/media | required | This directory contains a theme preview (a screenshot of your theme).

It should not be required as admin theme do not have it and also in [Create a new storefront theme page](https://devdocs.magento.com/guides/v2.2/frontend-dev-guide/themes/theme-create.htmll)
it has line

> If you do not have a preview image for your theme, remove the <media> node:

Under Static view files section there are some quotes escaped:

> ├── css/ (except the \&quot;source\&quot; sub-directory)

### PR DESCRIPTION
**Pull request by gelanivishal**
## This PR is a:

- [ ] New topic
- [X] Content update
- [ ] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary

Fixes #3164 

When this pull request is merged, it will improve the theme structure documentation and clarify theme related issues as per #3164. 

## Additional information

List all affected URLs 

- https://devdocs.magento.com/guides/v2.2/frontend-dev-guide/themes/theme-structure.html

### CODE REVIEW

**Comment by magento-cicd2** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/frontend-dev-guide/themes/theme-structure.md [hunk 1]
```diff
@@ -77,13 +77,13 @@ The directories and files structure described below is the most extended one. It
...
       <td colspan="1">
-        <code>/&lt;Vendor&gt;_&lt;Module&gt;/web/css/source</code>
+        <code>/&lt;Vendor&gt;_&lt;Module&gt;/web/css/</code>
       </td>
       <td colspan="1">
         optional
       </td>
       <td colspan="1">
-          Module-specific styles (<code>.css</code> and/or <code>.less</code> files). General styles for the module are in the <code>_module.less</code> file, and styles for widgets are in <code>_widgets.less</code>.
+          Module-specific styles (<code>.css</code> and/or <code>.less</code> files). General styles for the module are in the <code>_module.less</code> file, and styles for widgets are in <code>_widgets.less</code><br/><br/>Ex: [Module_Theme]({{ site.mage2300url }}app/code/Magento/Theme/view/frontend/web/css) or description field should be corrected</code>.
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/themes/theme-structure.md]: The `guides/v2.1/frontend-dev-guide/themes/theme-structure.md` file is symlinked to the v2.2 and v2.3 files, so if you really want to add a link to the magento code repo here using the `{{ site.mage2300url }}` variable, you must remove the symlinks, create new files for 2.2 and 2.3, and add the version-specific variable to each file just like #3423.

File: guides/v2.1/frontend-dev-guide/themes/theme-structure.md [hunk 2]
```diff
@@ -152,7 +152,7 @@ The directories and files structure described below is the most extended one. It
       </td>
       <td colspan="1">required</td>
       <td colspan="1">
-        This directory contains a theme preview (a screenshot of your theme).
+        This directory contains a theme preview (a screenshot of your theme). It should not be required as admin theme do not have it and also in [Create a new storefront theme page](https://devdocs.magento.com/guides/v2.2/frontend-dev-guide/themes/theme-create.html#fedg_create_theme_how-to_declare).
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/themes/theme-structure.md]: If this isn't required, then we should simply change "required" to "optional" in the preceding cell and remove this text.

File: guides/v2.1/frontend-dev-guide/themes/theme-structure.md [hunk 3]
```diff
@@ -152,7 +152,7 @@ The directories and files structure described below is the most extended one. It
       </td>
       <td colspan="1">required</td>
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/themes/theme-structure.md]: change to "optional"

File: guides/v2.1/frontend-dev-guide/themes/theme-structure.md [hunk 4]
```diff
@@ -150,9 +150,9 @@ The directories and files structure described below is the most extended one. It
       <td colspan="1">
         <code>/media</code>
       </td>
-      <td colspan="1">required</td>
+      <td colspan="1">Optional</td>
       <td colspan="1">
-        This directory contains a theme preview (a screenshot of your theme).
+        This directory contains a theme preview (a screenshot of your theme). It should not be required as admin theme do not have it and also in [Create a new storefront theme page](https://devdocs.magento.com/guides/v2.2/frontend-dev-guide/themes/theme-create.html#fedg_create_theme_how-to_declare).
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/themes/theme-structure.md]:
```suggestion
        This directory contains a theme preview (a screenshot of your theme).
```

**Comment by gelanivishal** [comment]
@jeff-matthews I have performed as requested changes.

File: guides/v2.1/frontend-dev-guide/themes/theme-structure.md [hunk 5]
```diff
@@ -77,13 +77,13 @@ The directories and files structure described below is the most extended one. It
...
       <td colspan="1">
-        <code>/&lt;Vendor&gt;_&lt;Module&gt;/web/css/source</code>
+        <code>/&lt;Vendor&gt;_&lt;Module&gt;/web/css/</code>
       </td>
       <td colspan="1">
         optional
       </td>
       <td colspan="1">
-          Module-specific styles (<code>.css</code> and/or <code>.less</code> files). General styles for the module are in the <code>_module.less</code> file, and styles for widgets are in <code>_widgets.less</code>.
+          Module-specific styles (<code>.css</code> and/or <code>.less</code> files). General styles for the module are in the <code>_module.less</code> file, and styles for widgets are in <code>_widgets.less</code><br/><br/>Ex: [Module_Theme]({{ site.mage2100url }}app/code/Magento/Theme/view/frontend/web/css) or description field should be corrected</code>.
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/themes/theme-structure.md]:
```suggestion
          Module-specific styles (<code>.css</code> and/or <code>.less</code> files). General styles for the module are in the <code>_module.less</code> file, and styles for widgets are in <code>_widgets.less</code><br/><br/>Ex: [Module_Theme]({{ site.mage2100url }}app/code/Magento/Theme/view/frontend/web/css) or description field should be corrected.
```

File: guides/v2.1/frontend-dev-guide/themes/theme-structure.md (hunk 1 above)
Comment (jeff-matthews) [reply guides/v2.1/frontend-dev-guide/themes/theme-structure.md]: @gelanivishal, you still need to create unique files for 2.2 and 2.3 if you want to use the `{{site.mage2100url}}` var.

Comment (jeff-matthews) [reply guides/v2.1/frontend-dev-guide/themes/theme-structure.md]: @gelanivishal, if we publish this as-is, the output will create 2 different versions of the topic (2.1, 2.2, and 2.3), but all three will link to the 2.3 branch of magento/magento2. Each version of the docs should link to the correct version of the code repo, that's why I'm asking you to delete symlinks and create new files if you insist on including this new version-specific URL.

**Comment by gelanivishal** [comment]
@jeff-matthews Is there anything remaining?

**Comment by jeff-matthews** [comment]
@gelanivishal, yes. If you want to link to the magento/magento2 code repo in the docs using the `{{ site.mage2300url }}` variable, you must delete the symlinks from 2.1 to 2.2 and 2.3, create new files, and add the appropriate version-specific variable for each version.

**Comment by gelanivishal** [comment]
@jeff-matthews I got it and also followed to remove symlinks but it does not work for you. So can you please remove symlinks for 2.2 and 2.3 version and commit it then I'll change content on it.

**Comment by jeff-matthews** [comment]
Sure @gelanivishal, I'll take care of this for you. See https://github.com/magento/devdocs/pull/3494

**Pull request closed without merging**